src/MongoDB/ReadPreference.c

```diff
@@ -269,6 +269,12 @@ static PHP_METHOD(ReadPreference, __construct)
 		if (Z_TYPE_P(hedge) == IS_ARRAY || Z_TYPE_P(hedge) == IS_OBJECT) {
 			bson_t* hedge_doc = bson_new();
 
+			if (mongoc_read_prefs_get_mode(intern->read_preference) == MONGOC_READ_PRIMARY) {
+				phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT, "hedge may not be used with primary mode");
+				bson_destroy(hedge_doc);
+				return;
```

Comment:
Does this belong in the other commit? This commit is for `maxStalenessSeconds`.

Reply:
You are correct. Fixed.